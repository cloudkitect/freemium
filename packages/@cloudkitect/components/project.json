{
  "name": "@cloudkitect/components",
  "root": "packages/@cloudkitect/components",
  "targets": {
    "default": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen default",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "pre-compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen pre-compile",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen compile",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "post-compile": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen post-compile",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "test": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen test",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "package": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "build": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen build",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "test:watch": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen test:watch",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "bump": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen bump",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "unbump": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen unbump",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "release": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen release",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "post-upgrade": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen post-upgrade",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "upgrade": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen upgrade",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "watch": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen watch",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "eslint": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen eslint",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "compat": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen compat",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "package-all": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package-all",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "package:js": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package:js",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "package:python": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen package:python",
        "cwd": "packages/@cloudkitect/components"
      }
    },
    "docgen": {
      "executor": "nx:run-commands",
      "options": {
        "command": "pnpm exec projen docgen",
        "cwd": "packages/@cloudkitect/components"
      }
    }
  },
  "//": "~~ Generated by projen. To modify, edit .projenrc.js and run \"npx projen\"."
}
