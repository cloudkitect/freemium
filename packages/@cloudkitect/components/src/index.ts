export * from './ck-certificatemanager/ck-dns-validated-certificate';
export * from './ck-cloudfront/ck-distribution';
export * from './ck-core/ck-vendor-tags';
export * from './ck-core/ck-tag';
export * from './ck-core/ck-utils';
export * from './ck-core/ck-region';
export * from './ck-dynamodb/ck-table-v2';
export * from './ck-ec2/ck-vpc';
export * from './ck-ecr/ck-repository';
export * from './ck-ecs/ck-cluster';
export * from './ck-ecs/ck-fargate-clsuter';
export * from './ck-ecs/ck-fargate-service';
export * from './ck-ecs/ck-fargate-task-definition';
export * from './ck-elasticloadbalancingv2/ck-public-application-loadbalancer';
export * from './ck-s3/ck-bucket';
export * from './ck-stack/ck-stack';
