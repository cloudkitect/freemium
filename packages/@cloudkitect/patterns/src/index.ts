export * from './ck-website/ck-serverless-website';
export * from './ck-container/ck-serverless-container-app';