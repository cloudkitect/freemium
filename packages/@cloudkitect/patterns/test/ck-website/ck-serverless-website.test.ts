import { CkAccountType, CkStack, CkStackProps } from '@cloudkitect/components';
import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { CkServerlessWebsite } from '../../src';

let app: App;
let stack: CkStack;
const stackProps: CkStackProps = {
  ckAccountType: CkAccountType.PRODUCTION,
  ckApplication: 'Test',
  ckCompany: 'CloudKitect',
  env: { account: '123', region: 'us-east-1' },
};

describe('Well Architected Serverless Website', () => {
  beforeEach(() => {
    app = new App();
    stack = new CkStack(app, 'TestStack', stackProps);
  });

  it('CP-73: Default website configuration', () => {
    new CkServerlessWebsite(stack, 'Website', {
      ckPathToContent: './test/site-content',
      ckDomainName: 'cloudkitect.com',
    });

    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });

});
