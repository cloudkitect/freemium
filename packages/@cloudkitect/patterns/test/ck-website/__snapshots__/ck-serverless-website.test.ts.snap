// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Well Architected Serverless Website CP-73: Default website configuration 1`] = `
{
  "Mappings": {
    "AWSCloudFrontPartitionHostedZoneIdMap": {
      "aws": {
        "zoneId": "Z2FDTNDATAQYW2",
      },
      "aws-cn": {
        "zoneId": "Z3RFFRIM2A3IF5",
      },
    },
  },
  "Outputs": {
    "WebsiteUrlOutput03C89C484": {
      "Value": "cloudkitect.com",
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C1024MiB1024MiB6D9F8D19": {
      "DependsOn": [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C1024MiB1024MiBServiceRoleDefaultPolicy2360480C",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C1024MiB1024MiBServiceRoleBE855947",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-123-us-east-1",
          "S3Key": "2d56e153cac88d3e0c2f842e8e6f6783b8725bf91f95e0673b4725448a56e96d.zip",
        },
        "Environment": {
          "Variables": {
            "AWS_CA_BUNDLE": "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
          },
        },
        "EphemeralStorage": {
          "Size": 1024,
        },
        "Handler": "index.handler",
        "Layers": [
          {
            "Ref": "WebsiteDeployWithInvalidationAwsCliLayerCCF1A2FF",
          },
        ],
        "MemorySize": 1024,
        "Role": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C1024MiB1024MiBServiceRoleBE855947",
            "Arn",
          ],
        },
        "Runtime": "python3.9",
        "Tags": [
          {
            "Key": "CkComponents",
            "Value": "2.128.0",
          },
          {
            "Key": "CkVendor",
            "Value": "CloudKitect",
          },
        ],
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C1024MiB1024MiBServiceRoleBE855947": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "CkComponents",
            "Value": "2.128.0",
          },
          {
            "Key": "CkVendor",
            "Value": "CloudKitect",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C1024MiB1024MiBServiceRoleDefaultPolicy2360480C": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::cdk-hnb659fds-assets-123-us-east-1",
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::cdk-hnb659fds-assets-123-us-east-1/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "WebsiteWebsiteBucket222EFFC9",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "WebsiteWebsiteBucket222EFFC9",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "cloudfront:GetInvalidation",
                "cloudfront:CreateInvalidation",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C1024MiB1024MiBServiceRoleDefaultPolicy2360480C",
        "Roles": [
          {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C1024MiB1024MiBServiceRoleBE855947",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "WebsiteCdn68936C4F": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "DistributionConfig": {
          "Aliases": [
            "cloudkitect.com",
            "www.cloudkitect.com",
          ],
          "CustomErrorResponses": [
            {
              "ErrorCachingMinTTL": 1800,
              "ErrorCode": 403,
              "ResponseCode": 403,
              "ResponsePagePath": "/error.html",
            },
          ],
          "DefaultCacheBehavior": {
            "AllowedMethods": [
              "GET",
              "HEAD",
              "OPTIONS",
            ],
            "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
            "Compress": true,
            "TargetOriginId": "TestStackWebsiteCdnOrigin11A1516B8",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": [
            {
              "DomainName": {
                "Fn::GetAtt": [
                  "WebsiteWebsiteBucket222EFFC9",
                  "RegionalDomainName",
                ],
              },
              "Id": "TestStackWebsiteCdnOrigin11A1516B8",
              "S3OriginConfig": {
                "OriginAccessIdentity": {
                  "Fn::Join": [
                    "",
                    [
                      "origin-access-identity/cloudfront/",
                      {
                        "Ref": "WebsiteOAIA6F916E0",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "PriceClass": "PriceClass_All",
          "ViewerCertificate": {
            "AcmCertificateArn": {
              "Ref": "WebsiteCert9FAB4E0E",
            },
            "MinimumProtocolVersion": "TLSv1.2_2021",
            "SslSupportMethod": "sni-only",
          },
        },
        "Tags": [
          {
            "Key": "CkComponents",
            "Value": "2.128.0",
          },
          {
            "Key": "CkVendor",
            "Value": "CloudKitect",
          },
        ],
      },
      "Type": "AWS::CloudFront::Distribution",
      "UpdateReplacePolicy": "Retain",
    },
    "WebsiteCert9FAB4E0E": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "DomainName": "www.cloudkitect.com",
        "DomainValidationOptions": [
          {
            "DomainName": "www.cloudkitect.com",
            "HostedZoneId": "DUMMY",
          },
          {
            "DomainName": "cloudkitect.com",
            "HostedZoneId": "DUMMY",
          },
        ],
        "SubjectAlternativeNames": [
          "cloudkitect.com",
        ],
        "Tags": [
          {
            "Key": "CkComponents",
            "Value": "2.128.0",
          },
          {
            "Key": "CkVendor",
            "Value": "CloudKitect",
          },
          {
            "Key": "Name",
            "Value": "TestStack/Website/Cert",
          },
        ],
        "ValidationMethod": "DNS",
      },
      "Type": "AWS::CertificateManager::Certificate",
      "UpdateReplacePolicy": "Retain",
    },
    "WebsiteCfAliasRecord88E7BA4E": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "WebsiteCdn68936C4F",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "www.cloudkitect.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
      "UpdateReplacePolicy": "Retain",
    },
    "WebsiteDeployWithInvalidationAwsCliLayerCCF1A2FF": {
      "Properties": {
        "Content": {
          "S3Bucket": "cdk-hnb659fds-assets-123-us-east-1",
          "S3Key": "3322b7049fb0ed2b7cbb644a2ada8d1116ff80c32dca89e6ada846b5de26f961.zip",
        },
        "Description": "/opt/awscli/aws",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "WebsiteDeployWithInvalidationCustomResource1024MiB1024MiBB3990B34": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "DestinationBucketName": {
          "Ref": "WebsiteWebsiteBucket222EFFC9",
        },
        "DistributionId": {
          "Ref": "WebsiteCdn68936C4F",
        },
        "DistributionPaths": [
          "/*",
        ],
        "Prune": true,
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C1024MiB1024MiB6D9F8D19",
            "Arn",
          ],
        },
        "SourceBucketNames": [
          "cdk-hnb659fds-assets-123-us-east-1",
        ],
        "SourceObjectKeys": [
          "07bf4924b51e604006ba723d9d6b5bda96d7e9d0e634c59ef2292b6480656b10.zip",
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
    "WebsiteOAIA6F916E0": {
      "Properties": {
        "CloudFrontOriginAccessIdentityConfig": {
          "Comment": "OAI for cloudkitect.com",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "WebsiteWebsiteBucket222EFFC9": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "BucketName": "www.cloudkitect.com",
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "Tags": [
          {
            "Key": "aws-cdk:cr-owned:aa4297a8",
            "Value": "true",
          },
          {
            "Key": "CkComponents",
            "Value": "2.128.0",
          },
          {
            "Key": "CkVendor",
            "Value": "CloudKitect",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "WebsiteWebsiteBucketPolicy501B6CDA": {
      "Properties": {
        "Bucket": {
          "Ref": "WebsiteWebsiteBucket222EFFC9",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": {
                "AWS": "*",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "WebsiteWebsiteBucket222EFFC9",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "WebsiteWebsiteBucket222EFFC9",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": {
                "CanonicalUser": {
                  "Fn::GetAtt": [
                    "WebsiteOAIA6F916E0",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "WebsiteWebsiteBucket222EFFC9",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "WebsiteWwwAliasRecord05D4EF22A": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Ref": "WebsiteCfAliasRecord88E7BA4E",
          },
          "HostedZoneId": "DUMMY",
        },
        "HostedZoneId": "DUMMY",
        "Name": "cloudkitect.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
